espresso-server/app/src/test/java/io/appium/espressoserver/test/helpers/w3c/TickTest.java

```diff
@@ -0,0 +1,278 @@
+        pointerInputState.setY(6L);
+        pointerInputState.addPressed(0);
+        pointerInputState.addPressed(1);
+        inputStateTable.addInputState(sourceId, pointerInputState);
+        inputStateTable.addInputState(sourceId2, pointerInputState);
+
+        // Construct pointer move event
+        ActionObject actionObjectOne = new ActionObject(sourceId, POINTER, null, 0);
+        actionObjectOne.setSubType(POINTER_MOVE);
+        actionObjectOne.setPointer(TOUCH);
+        actionObjectOne.setX(10L);
+        actionObjectOne.setY(20L);
+        actionObjectOne.setOrigin(VIEWPORT);
+
+        // Construct another pointer move event
+        ActionObject actionObjectTwo = new ActionObject(sourceId2, POINTER, null, 0);
+        actionObjectTwo.setSubType(POINTER_MOVE);
+        actionObjectTwo.setPointer(TOUCH);
+        actionObjectTwo.setX(10L);
+        actionObjectTwo.setY(20L);
+        actionObjectTwo.setOrigin(VIEWPORT);
+
+        // Add two pointer move actions to verify that they can run on multiple threads separately
+        tick.addAction(actionObjectOne);
+        tick.addAction(actionObjectTwo);
+
+        class ExtendedDummyW3CActionAdapter extends DummyW3CActionAdapter {
+            @Override
+            public void pointerMove(String sourceIdCalled,
+                                    PointerType pointerType,
+                                    long currentX, long currentY,
+                                    long x, long y,
+                                    Set<Integer> buttons,
+                                    KeyInputState globalKeyInputState) throws AppiumException {
+                assertEquals(pointerType, TOUCH);
+                assertEquals(currentX, 5L);
+                assertEquals(currentY, 6L);
+                assertEquals(x, 10L);
+                assertEquals(y, 20L);
+                assertTrue(buttons.contains(0));
+                assertTrue(buttons.contains(1));
+                assertTrue(globalKeyInputState.isShift());
+            }
+
+            @Override
+            public void keyDown(KeyEvent keyDownEvent) {
+            }
+        };
+
+        W3CActionAdapter baseW3CActionAdapter = new ExtendedDummyW3CActionAdapter();
+
+        List<Callable<Void>> callables = tick.dispatch(baseW3CActionAdapter, inputStateTable, tick.calculateTickDuration());
+        Executor executor = Executors.newFixedThreadPool(callables.size());
+        CompletionService<Void> completionService = new ExecutorCompletionService<>(executor);
+        for(Callable<Void> callable:callables) {
+            completionService.submit(callable);
+        }
+
+        int received = 0;
+        boolean errors = false;
+        while (received < callables.size() && !errors) {
+            Future<Void> resultFuture = completionService.take(); //blocks if none available
+            try {
+                resultFuture.get();
+                received ++;
+            }
+            catch(Exception e) {
```

Comment:
Why exceptions should be caught in a test?

Reply:
Good point, I could probably just not count the errors and let it throw the exception so that the test fails.